python/adelphi/bin/adelphi

```diff
@@ -103,104 +151,119 @@ def unique(iterable, key=None):
-
-
-def validate_keyspaces(keyspace_objs):
-    if len(keyspace_objs) == 0:
-        log.info("No keyspaces selected")
+    if len(keyspaces) == 0:
+        log.info("No valid keyspaces selected")
         exit(1)
 
+    log.info("Processing the following keyspaces: %s", ','.join((ks.name for ks in keyspaces)))
+    return {anonymize_keyspace(ks) if props['anonymize'] else ks : build_keyspace_id(ks) for ks in keyspaces}
+
 
 def get_cluster_metadata(cluster):
     hosts = cluster.metadata.all_hosts()
     unique_dcs = unique((host.datacenter for host in hosts))
-    return {"host_count": len(hosts), "dc_count": sum(1 for i in unique_dcs)}
-
-
-def gather_keyspaces(ctx):
-    partial_fn = partial(get_keyspaces, ctx)
-    keyspaces = with_cluster(partial_fn, ctx)
-
-    log.info("Exporting schema for keyspaces %s", ','.join((k.name for k in keyspaces)))
+    return {"host-count": len(hosts), "dc-count": sum(1 for i in unique_dcs)}
 
-    validate_keyspaces(keyspaces)
-    return keyspaces
 
+def build_keyspace_map_and_metadata(props):
+    def build_fn(cluster):
+        metadata = {k : props[k] for k in ["purpose", "maturity"] if k in props}
+        metadata.update(get_cluster_metadata(cluster))
+        metadata["creation_timestamp"] = datetime.now(utc).isoformat()
+        return (get_keyspaces(props, cluster), {k : metadata[k] for k in metadata.keys() if metadata[k]})
 
-def gather_metadata(ctx):
-    metadata = {k : ctx.obj[k] for k in ["purpose", "maturity"]}
-    metadata.update(with_cluster(get_cluster_metadata, ctx))
-    return {k : metadata[k] for k in metadata.keys() if metadata[k] is not None}
+    return with_cluster(build_fn, props)
 
 
-def export_keyspaces(ctx, export_fn, keyspace_objs, metadata, output_dir = None):
-    options = {'rf': ctx.obj['rf']}
-    export_dir = output_dir or ctx.obj["output-dir"]
+def export_keyspaces(props, export_fn, keyspace_map, metadata):
+    options = {'rf': props['rf']}
+    export_dir = props["output-dir"]
 
     if export_dir is None:
-        print(export_fn(keyspace_objs, metadata, options))
+        print(export_fn(keyspace_map.keys(), metadata, options))
     else:
-        for keyspace_obj in keyspace_objs:
-            with open(os.path.join(export_dir, keyspace_obj.name),'w') as output_file:
+        for (keyspace_obj, keyspace_id) in keyspace_map.items():
+            # Output filename should use the file-safe alphabet for b64 encoding
+            output_file_name = keyspace_id if props["anonymize"] else keyspace_obj.name
+            with open(os.path.join(export_dir, output_file_name),'w') as output_file:
                 log.info("Writing schema for keyspace " + keyspace_obj.name + " to file " + output_file.name)
+                metadata['keyspace_id'] = keyspace_id
                 output_file.write(export_fn([keyspace_obj], metadata, options))
+                del metadata['keyspace_id']
```

Comment:
I prefer the name "keyspace_id" here and converted to it throughout this PR, not just in metadata but also in code.  It moves away from a name based on the impl (the various "hash" names that were used before) and identifies the field based on it's function in the code, which seems better on it's face.